### PR TITLE
DAOS-9602 chk: verify container label

### PR DESCRIPTION
**Pull request by Nasf-Fan**
If the container label in the container service (cont_svc::cs_uuids)
exists but does not match the label in the container property, then
trust the container service and reset the one in container property.

Otherwise, if the container label in the container service does not
exists, but the one in the container property is there, then trust
the label in the container property and add it to container service.

Signed-off-by: Fan Yong <fan.yong@intel.com>

### CODE REVIEW

**Comment by Nasf-Fan** [comment]
target_auto_eviction failed for DAOS-11179, to be retested.

File: src/container/srv_container.c [hunk 1]
```diff
@@ -4537,3 +4537,120 @@ ds_cont_destroy_orphan(struct cont_svc *svc, uuid_t uuid)
...
+
+int
+ds_cont_iterate_labels(struct cont_svc *svc, rdb_iterate_cb_t cb, void *arg)
+{
+	struct rdb_tx	tx;
+	int		rc;
+
+	ABT_rwlock_rdlock(svc->cs_lock);
+	rc = rdb_tx_begin(svc->cs_rsvc->s_db, svc->cs_rsvc->s_term, &tx);
+	if (rc == 0) {
```
Comment (wangshilong) [inline src/container/srv_container.c:4549]: miss ABT_rwlock_unlock(svc->cs_lock); if rc != 0 ?

**Review by jolivier23 — APPROVED** [review]

File: src/container/srv_container.c (hunk 1 above)
Comment (Nasf-Fan) [reply src/container/srv_container.c:4549]: It is the wrong oder between lock and tx_begin. I will fix it. Thanks!

**Comment by Nasf-Fan** [comment]
@liw @wangshilong , please help to review the patch, thanks!

**Review by wangshilong — APPROVED** [review]

**Review by liw — APPROVED** [review]

**Pull request merged**